File: String/#383. Ransom Note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char,int>mp;
        for(auto i: magazine)
            mp[i]++;
        
        for(auto i: ransomNote){
            if(!mp.count(i) || mp[i]==0) return false;
            mp[i]--;
        }
        return true;
    }
};

File: Array/#383. Ransom Note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        freq = defaultdict(int)
        for e in magazine:
            freq[e] += 1
        for e in ransomNote:
            if freq[e] == 0:
                return False
            freq[e] -= 1

        return True